### PR TITLE
Handle localhost as ipv6 without bracket quoting

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

File: mockwebserver/src/test/java/okhttp3/mockwebserver/RecordedRequestTest.java [hunk 1]
```diff
@@ -102,4 +102,14 @@ private FakeSocket(InetAddress localAddress, int localPort, InetAddress remoteAd
 
     assertThat(request.getRequestUrl().toString()).isEqualTo("http://127.0.0.1/");
   }
+
+  @Test public void testUsesLocalIpv6WithoutBrackets() throws UnknownHostException {
```
Comment (monkey-mas) [inline mockwebserver/src/test/java/okhttp3/mockwebserver/RecordedRequestTest.java]: 👍 

nit: we might add a test for IPv4 as well? ;)
e.g. 
```
  @Test public void testLocalhostIpv4() throws UnknownHostException {
    Socket socket =
        new FakeSocket(InetAddress.getByAddress("localhost", new byte[] {127, 0, 0, 1}), 80);

    RecordedRequest request = new RecordedRequest("GET / HTTP/1.1", headers,
        Collections.emptyList(), 0, new Buffer(), 0, socket);

    assertThat(request.getRequestUrl().toString()).isEqualTo("http://localhost/");
  }
```

File: mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt [hunk 2]
```diff
@@ -101,7 +101,11 @@ class RecordedRequest(
       val inetAddress = socket.localAddress
 
       var hostname = inetAddress.hostName
-      if (inetAddress is Inet6Address) {
+      if (inetAddress is Inet6Address && hostname.contains(':')) {
+        // hostname is likely some form representing the ipv6 bytes
```
Comment (monkey-mas) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt]: super-nit: it'd be nice to write down a formal name, e.g. `IPv6` ? :)

**Comment by monkey-mas** [comment]
LGTM! :)

File: mockwebserver/src/test/java/okhttp3/mockwebserver/RecordedRequestTest.java (hunk 1 above)
Comment (yschimke) [reply mockwebserver/src/test/java/okhttp3/mockwebserver/RecordedRequestTest.java]: Thanks!

**Pull request merged**